app/knights.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+
+class Knights:
+    def __init__(
+            self,
+            name: str,
+            power: int,
+            hp: int,
+            armour: list[dict],
+            weapon: dict,
+            potion: dict | None,

```

Comment:
The `potion` attribute is expected to be a dictionary with an 'effect' key. Consider adding a type check or validation to ensure that `potion` is either `None` or a dictionary with the expected structure to prevent runtime errors.